skbeam/core/accumulators/arrayProjector.py

```diff
@@ -0,0 +1,155 @@
+        if weights is None:
+            self._normalization_array = (np.bincount( self._bin_assignments.flatten() ) \
+                                             + 1e-100).astype(np.float)
+        else:
+            self._normalization_array = (np.bincount( self._bin_assignments.flatten(), weights=self.weights.flatten() ) \
+                                             + 1e-100).astype(np.float)
+
+        assert self.nbins >= self._bin_assignments.max() + 1, 'incorrect bin assignments'
+        self._normalization_array = self._normalization_array[:self.nbins]
+
+        return
+
+    def __call__(self, image):
+        """
+        Bin pixel intensities.
+        
+        Parameters
+        ----------            
+        image : np.ndarray
+            The intensity at each pixel
+        Returns
+        -------
+        bin_values : np.ndarray
+            The average intensity in the bin
+        """
+
+        if not (image.shape == self.bin_values.shape):
+            raise ValueError('`image` and `bin_values` must have the same shape',image.shape,self.bin_values.shape)
+
+        if self.weights is None:
+            weights = image.flatten()
+        else:
+            weights = image.flatten() * self.weights.flatten()
+            if not (image.shape == self.weights.shape):
+                raise ValueError('`image` and `weights` must have the same shape')
+
+        bin_values = np.bincount(self._bin_assignments.flatten(), weights=weights)
+
+        if self.norm:
+            bin_values /= self._normalization_array
+
+        assert bin_values.shape[0] == self.nbins
+
+        return bin_values
+
+    @property
+    def bin_centers(self):
+        """
+        Returns:
+        --------
+        bin_centers : ndarray, float
+            The center of each bin.
+        """
+        return (np.arange(self.nbins) + 0.5) * self.bin_width + self.bin_values.min()
+
+class RadialProjector(ArrayProjector) :
+    """ 
+    Project a 2D image onto a radial axis
+    """
+    
+    def __init__(self, rows, columns, xc=None, yc=None, rmin=None, rmax=None, phimin=None, phimax=None, nbins=101, weights=None, norm=True):
+        """
+        Parameters:
+        -----------
+        rows,columns:  shape of image to be projected
+        xc,yc:         location (in pixels) of origin (default: center of image)
+        rmin,rmax:     radial range to include in projection, in pixels (default: no limits)
+        phimin,phimax: phi range to include in projection, in degrees (default: no limits)
+        nbins:         number of bins in projected histogram
+        weights:       np.ndarray.  weight to be applied to each pixel in image.  this can
+                       be used as a mask if the weight is set to zero.
+        norm:          boolean indicating whether bin entries in the projected histogram should be divided
+                       by weights (number of pixels, in the case where the weights are 1).
```

Comment:
Sphinx is not going to render this docstring correctly.  The rest of the docstrings look good though.